### PR TITLE
Testing Custom Enumerable with RSpec

### PR DESCRIPTION
**Pull request by mohammadumar28**
- [x] Add Test Case for all the Enumerable Methods

- [x] Resolve all the Rubocop Errors

- [x] Add Edge Cases

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+  let(:array) { Array.new(SIZE) { rand(0..SIZE) } }
+  let(:range) { Range.new(0, 9) }
+  let(:hash) { { a: 'A', b: 'B', c: 'C' } }
+  let(:numbers) { [1, 2.332, 66, 90.7] }
+  let(:words) { %w[cato dog lion dolphin] }
+  let(:block) { proc { |el| el } }
+  let(:block_num) { proc { |el| el < 5 } }
+  let(:block_args) { proc { |el, index| el + index } }
+  # let(:block_acc) { proc { |sum, el| sum + el } }
+
```
Comment (addod19) [inline spec/enumerable_spec.rb:14]: Kindly remove commented code!! :slightly_smiling_face:

**Comment by Dandush03** [comment]
# Status: 🚀 Approved 🚀 

## Great Job! Keep it up

# Happy Coding 🤓 🤓
